components/ProfilePage/ProfilePage.tsx

```diff
@@ -39,13 +40,19 @@ export function ProfilePage(profileprops: {
     profile?.role === "pendingUpgrade" ||
     false
 
+  const [isProfilePublic, setIsProfilePublic] = useState<boolean>(false)
+
+  useEffect(() => {
+    setIsProfilePublic(profile?.public ? profile.public : false)
```

Comment:
Pretty sure this is equivalent?
```suggestion
    setIsProfilePublic(profile?.public)
```

Reply:
I was worried about letting an undefined value into "setIs..." but it turns out that you are correct and it's redundant;

updated to per your code